### PR TITLE
moved shared functions to base class

### PR DESCRIPTION
**Pull request by kayernyc**
moved some functions into the base class

### CODE REVIEW

File: src/views/DatePickerBaseView.ts [hunk 1]
```diff
@@ -1,34 +1,69 @@
...
     this.frameElement.className += " date-picker-view";
     return this.frameElement;
   }
 
-  abstract append(parentElement: HTMLElement): void;
+  append(parentElement: HTMLElement): void {
+    this.frameElement = this.initFrameView(this.continuousScroll);
+    const classNames = this.frameElement.className.split("");
+    classNames.push(this.frameElementClassName);
+    this.frameElement.className = classNames.join(" ");
```
Comment (deadnancy) [inline src/views/DatePickerBaseView.ts:40]: Hello! I am missing something here. When I change the date picker control to a type that uses this block, I'm seeing `this.frameElement.className` end up as `d a t e - p i c k e r - v i e w date-picker-list`. What's the reasoning behind splitting `this.frameElement.className`into individual letters?

Comment (kayernyc) [reply src/views/DatePickerBaseView.ts:40]: Hi! I'm not able to replicate this. Can you tell me what you did?

Comment (deadnancy) [reply src/views/DatePickerBaseView.ts:40]: Changed `dateType` to `1` in index.ts, then sent the output of `this.frameElement.className` to console after line 40.

Comment (kayernyc) [reply src/views/DatePickerBaseView.ts:40]: Awesome catch!!! Thank you!

**Pull request merged**